google/cloud/storage/internal/async/object_descriptor_impl.h

```diff
@@ -77,7 +79,9 @@ class ObjectDescriptorImpl
     return CopyActiveRanges(std::unique_lock<std::mutex>(mu_));
   }
 
-  auto CurrentStream(std::unique_lock<std::mutex>) const { return stream_; }
+  auto CurrentStream(std::unique_lock<std::mutex>) const {
+    return streams_[active_stream_];
```

Comment:
It looks like this is always `streams_.back()`, and therefore that `active_stream_` is redundant.